sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java

```diff
@@ -34,31 +36,33 @@
 
   private final ConcurrentMap<String, Connection> connections = new ConcurrentHashMap<>();
 
-  private final DataStore dataStore;
+  private final DataStoreFactory dataStoreFactory;
   private final AuthenticationService authenticator;
 
-  public StargateMeta(DataStore dataStore, AuthenticationService authenticator) {
-    this.dataStore = dataStore;
+  public StargateMeta(DataStoreFactory dataStoreFactory, AuthenticationService authenticator) {
+    this.dataStoreFactory = dataStoreFactory;
     this.authenticator = authenticator;
   }
 
   private void newConnection(ConnectionHandle ch, Map<String, String> info) {
     connections.computeIfAbsent(
         ch.id,
         sid -> {
-          String username = info.get("user");
+          String username;
           String password = info.get("password");
-          if (username == null || password == null) {
+          if (password == null) {
             throw new IllegalArgumentException("Missing credentials in connection properties.");
           }
 
           try {
-            authenticator.createToken(username, password);
+            StoredCredentials credentials = authenticator.validateToken(password);
```

Comment:
Don't we want this to support both? Similar to how cql module works?

Reply:
CQL validates passwords through SPNEGO, IIRC... but it does not look like it maps well to JDBC... there's no API to negotiate with the real client, AFAIK... We could simulate that on the server side, of course.

Reply:
Ah, CQL uses SASL, but still it does not map well to JDBC :)

Reply:
I'd prefer to defer actual password validation if tokens are ok short term. WDYT?

Reply:
Sorry that should have been phrased better. I meant don't we want to support both traditional username/password auth _and_ token based auth?

Reply:
Another option is to add an new method to `AuthenticationService` to validate passwords without creating a token. This looks goable short term. Shall I open a PR to `master` for that?

Reply:
As discussed offline, we're going to defer password-validation for SQL connections until we have a core API in Stargate for password checks. Previously (before this PR) passwords were validated by _creating a token_, which looks like an overkill.